src/onepay-checkout.js

```diff
@@ -684,58 +660,54 @@ function prepareOnepayHttpRequestParams(params) {
-        let data = {};
-        try {
-          data = JSON.parse(httpRequest.responseText);
+function processOnepayHttpResponse(onepay, status, responseText) {
+  if (status === 200) {
+    let data = {};
+    try {
+      data = JSON.parse(responseText);
 
-          if (data !== null && 'ott' in data && 'occ' in data && 'amount' in data) {
-            onepay.total = data.amount;
-            onepay.occ = data.occ;
-            onepay.ott = data.ott;
-            onepay.qrBase64 = data.qrCodeAsBase64 || '';
-            onepay.externalUniqueNumber = data.externalUniqueNumber || '';
-
-            if (typeof Smartphone !== 'undefined') {
-              // If it's a mobile device, close the modal
-              if (Smartphone.isAny()) {
-                onepay.closeModal();
-              }
-
-              if (Smartphone.isAndroid()) {
-                Smartphone.androidContextChange(data.occ);
-                return;
-              }
-
-              if (Smartphone.isIOS()) {
-                Smartphone.iosContextChange(data.occ);
-                return;
-              }
-            }
-
-            onepay.countDownDate = new Date();
-
-            updateContentPayment(onepay);
-            let options = {'onepay': onepay};
-            getCredentials(options);
-          } else {
-            updateContentError(onepay);
-            console.log('Los datos recibidos no son los requeridos');
+      if (data !== null && 'ott' in data && 'occ' in data && 'amount' in data) {
+        onepay.total = data.amount;
+        onepay.occ = data.occ;
+        onepay.ott = data.ott;
+        onepay.qrBase64 = data.qrCodeAsBase64 || '';
+        onepay.externalUniqueNumber = data.externalUniqueNumber || '';
+
+        if (typeof Smartphone !== 'undefined') {
+          // If it's a mobile device, close the modal
+          if (Smartphone.isAny()) {
+            closeModal();
+          }
+
+          if (Smartphone.isAndroid()) {
+            Smartphone.androidContextChange(data.occ);
+            return;
+          }
+
+          if (Smartphone.isIOS()) {
+            Smartphone.iosContextChange(data.occ);
+            return;
           }
-        } catch (e) {
-          console.log('Falló el parseo de la respuesta');
-          console.log(e);
-          updateContentError(onepay);
         }
+
+        onepay.countDownDate = new Date();
+
+        updateContentPayment(onepay);
+        let options = {'onepay': onepay};
+        getCredentials(options);
       } else {
         updateContentError(onepay);
-        console.log('Hubo un problema con la solicitud HTTP: ' + httpRequest.responseText);
+        console.log('Los datos recibidos no son los requeridos');
       }
+    } catch (e) {
+      console.log('Falló el parseo de la respuesta');
+      console.log(e);
+      updateContentError(onepay);
     }
-  };
+  } else {
+    updateContentError(onepay);
+    console.log('Hubo un problema con la solicitud HTTP: ' + httpRequest.responseText);
```

Comment:
`httpRequest.responseText` is not valid here. It should be just `responseText`

Reply:
Fixed